friends/src/components/FriendForm.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+
+export default class FriendForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state={
+            friend: this.props.activeFriend || {
+                name: '',
+                age: '',
+                email: ''
+            }
+        }
+
+    }
+
+    componentDidUpdate(prevProps) {
+        if(
+            this.props.activeFriend &&
+            prevProps.activeFriend !== this.props.activeFriend
+        ) {
+            this.setState({
+                friend: this.props.activeFriend
+            })
+        }
+     }
+
+    changeHandler = e => {
+        e.persist();
+        this.setState(prevState => ({
+            friend: {
+                ...prevState.friend,
+                [e.target.name]: e.target.value
+                }
+        }))
+    };
+
+    handleSubmit = e => {
+        e.preventDefault();
+        this.props.addFriend(this.state.friend)
+    }
+
```

Comment:
At the time of reviewing this, update friend adds a new friend. To fix it, we just have to add another method that will handle the update by making use of your `updateFriend` method in `Friends.js`.
```suggestion
handleUpdate = e => {
    e.preventDefault();
    this.props.updateFriend(e, this.state.friend);
  };
```